src/formatDate.js

```diff
@@ -7,8 +7,34 @@
  *
  * @returns {string}
  */
+
+const getObjectDate = (dateParts, fromFormat) =>
+  fromFormat.reduce((objectFormat, format, i) => {
+    if (format === 'YY' && dateParts[i].length === 2) {
+      objectFormat['YYYY'] =
+        dateParts[i] < '30' ? `20${dateParts[i]}` : `19${dateParts[i]}`;
+    } else {
+      objectFormat[format] = dateParts[i];
+    }
+
+    return objectFormat;
+  }, {});
+
+const convertYear = (objectDate, toFormat) => {
+  if (!objectDate['YY'] && objectDate['YYYY'] && toFormat.includes('YY')) {
+    objectDate['YY'] = objectDate['YYYY'].slice(-2);
+  }
+};
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const separatorFrom = fromFormat.pop();
+  const separatorTo = toFormat.pop();
```

Comment:
Similarly to the previous comment, this line mutates the `toFormat` array by removing the last element. It's better to avoid mutating input arguments to prevent side effects.